src/createAgentManager.ts

```diff
@@ -109,36 +149,53 @@ export async function createAgentManager(agent: string | Agent, options: AgentMa
     const baseURL = options.baseURL || didApiUrl;
     const wsURL = options.wsURL || didSocketApiUrl;
     const abortController: AbortController = new AbortController();
+    const mxKey = options.mixpanelKey || mixpanelKey;
+
     const agentsApi = createAgentsApi(options.auth, baseURL);
     const ratingsAPI = createRatingsApi(options.auth, baseURL);
     const knowledgeApi = createKnowledgeApi(options.auth, baseURL);
 
     const agentInstance = typeof agent === 'string' ? await agentsApi.getById(agent) : agent;
     options.callbacks?.onAgentReady?.(agentInstance);
 
+    const analytics = AnalyticsProvider.getInstance({
+        mixPanelKey: mxKey,
+        agent: agentInstance,
+        ...options
+    });
+
     const socketManager = await SocketManager(options.auth, wsURL, options.callbacks.onChatEvents);
-    let { chat, streamingManager } = await initializeStreamAndChat(agentInstance, options, agentsApi);
+    let { chat, streamingManager } = await initializeStreamAndChat(agentInstance, options, agentsApi, analytics);
+
+    analytics.track('agent-sdk-created', getInitAnaliticsInfo(agentInstance));
 
     return {
         agent: agentInstance,
         chatId: chat.id,
         async reconnectToChat() {
+            analytics.track('agent-resume-chat');
             const { streamingManager: newStreamingManager } = await initializeStreamAndChat(
                 agentInstance,
                 options,
                 agentsApi,
+                analytics,
                 chat
             );
 
             streamingManager = newStreamingManager;
         },
         terminate() {
+            analytics.track('agent-chat', { event: 'terminated' });
             abortController.abort();
             socketManager.terminate();
 
             return streamingManager.terminate();
         },
         chat(messages: Message[]) {
+            analytics.track('agent-message-send', {
+                event: 'success',
+                messages: messages.length + 1,
+            });
```

Comment:
did we have a failure event in the original agent-ui code?
something like:
catch (e){
 analytics.track('agent-message-send', {event: 'error', reason: e.description
}

Reply:
Added missed mixpanel events for chat()